### PR TITLE
Bump dependencies, backfill capa

### PR DESCRIPTION
**Pull request by justaugustus**
**What this PR does / why we need it**:
- Updates capi version
- Updates Azure package versions
- Adds cluster-name label to machines (ref: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/567)
- Adds support for MachineClass (ref: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/571)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
Add support for MachineClass and `cluster-name` label to machines
```

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -42,25 +42,25 @@ required = [
 
 [[constraint]]
   name = "github.com/golang/mock"
-  version = "1.2.0"
+  branch = "master"
```
Comment (tariq1890) [inline Gopkg.toml]: can we point to `v1.2.0` ?

Comment (justaugustus) [reply Gopkg.toml]: I'm mimicking the capa repo here.

Comment (tariq1890) [reply Gopkg.toml]: But pointing to `master` is not advisable. I would encourage even the AWS provider to follow suit.

File: pkg/cloud/azure/actuators/machine_scope.go [hunk 2]
```diff
@@ -129,3 +132,44 @@ func (m *MachineScope) Close() {
...
+	if providerConfig.Value != nil {
+		klog.V(4).Info("Decoding ProviderConfig from Value")
+		return unmarshalProviderSpec(providerConfig.Value)
+	}
+
+	if providerConfig.ValueFrom != nil && providerConfig.ValueFrom.MachineClass != nil {
+		ref := providerConfig.ValueFrom.MachineClass
+		klog.V(4).Info("Decoding ProviderConfig from MachineClass")
+		klog.V(6).Infof("ref: %v", ref)
+		if ref.Provider != "" && ref.Provider != "aws" {
```
Comment (tariq1890) [inline pkg/cloud/azure/actuators/machine_scope.go]:
```suggestion
		if ref.Provider != "" && ref.Provider != "azure" {
```

File: Gopkg.toml (hunk 1 above)
Comment (justaugustus) [reply Gopkg.toml]: Fixed.

File: pkg/cloud/azure/actuators/machine_scope.go (hunk 2 above)
Comment (justaugustus) [reply pkg/cloud/azure/actuators/machine_scope.go]: Fixed.

**Comment by justaugustus** [comment]
@tariq1890 -- fixed up!

**Pull request merged**